ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/parser.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright © Aptos Foundation
+
+use crate::{db::upsert_uris, models::NFTMetadataCrawlerURIs, schema::nft_metadata_crawler_uris};
+use chrono::Utc;
+use diesel::{
+    r2d2::{ConnectionManager, PooledConnection},
+    PgConnection, QueryDsl, RunQueryDsl,
+};
+use google_cloud_auth::token_source::TokenSource;
+use image::{ImageBuffer, ImageFormat};
+use nft_metadata_crawler_utils::{
+    gcs::{write_image_to_gcs, write_json_to_gcs},
+    NFTMetadataCrawlerEntry,
+};
+use serde_json::Value;
+use std::error::Error;
+
+// Stuct that represents a parser for a single entry from queue
+pub struct Parser<'a> {
+    pub entry: NFTMetadataCrawlerEntry,
+    model: NFTMetadataCrawlerURIs,
+    format: ImageFormat,
+    bucket: String,
+    ts: &'a dyn TokenSource,
```

Comment:
why?

Reply:
To repeatedly get GCP service account auth bearer token in case it expires

Reply:
No i'm asking why the lifetime variable?